core/src/main/java/io/appflate/restmock/utils/RequestMatchers.java

```diff
@@ -66,6 +70,70 @@ protected boolean matchesSafely(RecordedRequest item) {
         };
     }
 
+  /**
+   * Creates a {@link RequestMatcher} object for determining if a {@link RecordedRequest} contains
+   * any query parameters.
+   *
+   * @return A new {@link RequestMatcher} object that will match a {@link RecordedRequest} if it
+   *         contains query parameters in its path.
+   */
+  public static RequestMatcher hasQueryParameters() {
+      return new RequestMatcher("matched query parameters") {
+        @Override
+        protected boolean matchesSafely(RecordedRequest item) {
+          try {
+            URL mockUrl = new URL("http", "localhost", item.getPath());
+            List<QueryParam> queryParams = RestMockUtils.splitQuery(mockUrl);
+            return queryParams.size() > 0;
+          } catch (MalformedURLException e) {
+            return false;
+          } catch (UnsupportedEncodingException e) {
+            return false;
+          }
+        }
+      };
+    }
+
+    /**
+     * Creates a {@link RequestMatcher} object for determining if a {@link RecordedRequest} contains
+     * a specific set of query parameters.
+     *
+     * @param expectedParams A set of {@link QueryParam} objects to be matched.
+     *
+     * @return A new {@link RequestMatcher} object that will match a {@link RecordedRequest} if it
+     *         contains the specified query parameters in its path. Note that this
+     *         {@link RequestMatcher} only matches the exact specification of query parameters. That
+     *         is, if any of the key/value pairs don't match, or if the number of expected parameters
+     *         doesn't match the number of actual parameters, this {@link RequestMatcher} will
+     *         return false.
+     */
+    public static RequestMatcher hasExactQueryParameters(final QueryParam... expectedParams) {
+      return new RequestMatcher("matched query parameters") {
+        @Override
+        protected boolean matchesSafely(RecordedRequest item) {
+          try {
+            URL mockUrl = new URL("http", "localhost", item.getPath());
+            List<QueryParam> actualParams = RestMockUtils.splitQuery(mockUrl);
+            if (actualParams.size() == 0 || actualParams.size() != expectedParams.length) {
+              return false;
+            }
+
+            for (QueryParam param : expectedParams) {
+              if (!actualParams.contains(param)) {
+                return false;
+              }
+            }
+
+            return true;
+          } catch (MalformedURLException e) {
+            return false;
```

Comment:
I can do this if you'd prefer, but the thing is that the only way this could happen is if `item.getPath()` returned a non-unicode string, which seems super unlikely, given that Java `String` objects are unicode by definition.